src/main/java/com/minecolonies/coremod/colony/managers/StatisticAchievementManager.java

```diff
@@ -0,0 +1,273 @@
+    private int felledTrees       = 0;
+    private int plantedSaplings   = 0;
+
+    /**
+     * List of achievements within the colony.
+     */
+    @NotNull
+    private final List<Advancement> colonyAchievements = new ArrayList<>();
+
+    @Override
+    public void readFromNBT(@NotNull final NBTTagCompound compound, @NotNull final Colony colony)
+    {
+        // Restore colony achievements
+        final NBTTagList achievementTagList = compound.getTagList(TAG_ACHIEVEMENT_LIST, Constants.NBT.TAG_COMPOUND);
+        for (int i = 0; i < achievementTagList.tagCount(); ++i)
+        {
+            final NBTTagCompound achievementCompound = achievementTagList.getCompoundTagAt(i);
+            final String achievementKey = achievementCompound.getString(TAG_ACHIEVEMENT);
+
+            //todo serialization
+            /*final StatBase statBase = StatList.getOneShotStat(achievementKey);
+             if (statBase instanceof Advancement)
+            {
+                colonyAchievements.add((Advancement) statBase);
+            }*/
+        }
+
+        //Statistics
+        final NBTTagCompound statisticsCompound = compound.getCompoundTag(TAG_STATISTICS);
+        final NBTTagCompound minerStatisticsCompound = statisticsCompound.getCompoundTag(TAG_MINER_STATISTICS);
+        final NBTTagCompound farmerStatisticsCompound = statisticsCompound.getCompoundTag(TAG_FARMER_STATISTICS);
+        final NBTTagCompound guardStatisticsCompound = statisticsCompound.getCompoundTag(TAG_FARMER_STATISTICS);
+        final NBTTagCompound builderStatisticsCompound = statisticsCompound.getCompoundTag(TAG_BUILDER_STATISTICS);
+        final NBTTagCompound fishermanStatisticsCompound = statisticsCompound.getCompoundTag(TAG_FISHERMAN_STATISTICS);
+        final NBTTagCompound lumberjackStatisticsCompound = statisticsCompound.getCompoundTag(TAG_LUMBERJACK_STATISTICS);
+        minedOres = minerStatisticsCompound.getInteger(TAG_MINER_ORES);
+        minedDiamonds = minerStatisticsCompound.getInteger(TAG_MINER_DIAMONDS);
+        harvestedCarrots = farmerStatisticsCompound.getInteger(TAG_FARMER_CARROTS);
+        harvestedPotatoes = farmerStatisticsCompound.getInteger(TAG_FARMER_POTATOES);
+        harvestedWheat = farmerStatisticsCompound.getInteger(TAG_FARMER_WHEAT);
+        killedMobs = guardStatisticsCompound.getInteger(TAG_GUARD_MOBS);
+        builtHuts = builderStatisticsCompound.getInteger(TAG_BUILDER_HUTS);
+        caughtFish = fishermanStatisticsCompound.getInteger(TAG_FISHERMAN_FISH);
+        felledTrees = lumberjackStatisticsCompound.getInteger(TAG_LUMBERJACK_TREES);
+        plantedSaplings = lumberjackStatisticsCompound.getInteger(TAG_LUMBERJACK_SAPLINGS);
+    }
+
+    @Override
+    public void writeToNBT(@NotNull final NBTTagCompound compound)
+    {
+        //  Achievements
+        @NotNull final NBTTagList achievementsTagList = new NBTTagList();
+        for (@NotNull final Advancement achievement : this.colonyAchievements)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused "achievement" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1481)